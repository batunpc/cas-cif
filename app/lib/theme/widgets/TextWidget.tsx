import { WidgetProps } from "@rjsf/core";
import Input from "@button-inc/bcgov-theme/Input";
import getRequiredLabel from "../utils/getRequiredLabel";

const TextWidget: React.FunctionComponent<WidgetProps> = ({
  id,
  placeholder,
  onChange,
  label,
  value,
  required,
  uiSchema,
}) => {
  return (
    <>
      <Input
        id={id}
        onChange={(e) => onChange(e.target.value || undefined)}
        placeholder={placeholder}
        label={getRequiredLabel(label, required)}
        value={value || ""}
        size={(uiSchema && uiSchema["bcgov:size"]) || "medium"}
        required={required}
      />
      <style jsx>
        {`
          :global(input) {
            width: 100%;
          }
        `}
      </style>
    </>
  );
};

export default TextWidget;
