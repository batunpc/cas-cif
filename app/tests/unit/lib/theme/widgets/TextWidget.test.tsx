import React from "react";
import TextWidget from "lib/theme/widgets/TextWidget";

import { render } from "@testing-library/react";

describe("The Text Widget", () => {
  it("Matches the snapshot with the value set", () => {
    const props: any = {
      id: "1",
      placeholder: "test placeholder",
      onChange: jest.fn(),
      label: "",
      value: "test",
      required: false,
    };

    const componentUnderTest = render(<TextWidget {...props} />);

    expect(componentUnderTest.container).toMatchSnapshot();
  });
});
