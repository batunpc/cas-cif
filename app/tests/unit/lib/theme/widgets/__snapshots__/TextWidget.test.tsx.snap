// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Text Widget Matches the snapshot with the value set 1`] = `
<div>
  <div
    class="sc-hKwDye jvzxmJ pg-input"
  >
    
    <input
      aria-label="1-input"
      class="sc-jRQBWg AhBXp pg-input-input"
      id="1"
      name="1-input"
      placeholder="test placeholder"
      value="test"
    />
  </div>
</div>
`;
